signac/contrib/project.py

```diff
@@ -2371,12 +2371,37 @@ def __eq__(self, other):
     def __len__(self):
         # Highly performance critical code path!!
         if self._filter or self._doc_filter:
-            # We use the standard function for determining job ids if and only if
-            # any of the two filter is provided.
+            # We use the standard function for determining job ids if a filter
+            # is provided.
             return len(self._project._find_job_ids(self._filter, self._doc_filter))
         else:
-            # Without filter we can simply return the length of the whole project.
-            return self._project.__len__()
+            # Without filters, we can simply return the length of the whole project.
+            return len(self._project)
+
+    def __contains__(self, job):
+        """Determine whether a job is in this cursor.
+
+        Parameters
+        ----------
+        job : :class:`~signac.contrib.job.Job`
+            The job to check.
+
+        Returns
+        -------
+        bool
+            True if the job matches the filter criteria and is initialized for this project.
+
+        """
+        if job not in self._project:
+            # Exit early if the job is not in the project.
+            return False
```

Comment:
Can you add a test for this? I feel like this test might catch a few bugs that could be in the code

Reply:
I added tests! Thanks for the reminder.